internal/policies/mount/adsys_mount/src/logger.rs

```diff
@@ -15,12 +15,14 @@ impl Log for Logger {
             return;
         }
 
-        eprintln!(
-            "{} - {}: {}",
-            Utc::now().format("%d/%m/%Y %H:%M:%S"),
-            record.level(),
-            record.args()
-        );
+        let fmt_time = match OffsetDateTime::now_utc().format(format_description!(
+            "[day]/[month]/[year] [hour]:[minute]:[second]"
+        )) {
+            Ok(t) => t,
+            Err(_) => String::from("00/00/0000 00:00:00"),
+        };
+
+        eprintln!("{} - {}: {}", fmt_time, record.level(), record.args());
```

Comment:
For this pattern, I recommend using a short-hand form:

```suggestion
        eprintln!(
            "{} - {}: {}",
            OffsetDateTime::now_utc()
                .format(format_description!(
                    "[day]/[month]/[year] [hour]:[minute]:[second]"
                ))
                .unwrap_or_else(|_| "00/00/0000 00:00:00".to_string()),
            record.level(),
            record.args()
        );
```

`unwrap_or_else` is lazily evaluated (meaning it will only run the closure code when the condition is `Err`), so don't worry about whether it will cause performance issues.

Reply:
Tbh, I think the shortened form is just less readable without achieving much else. In this cases, I'd rather go for readability instead of micro optimization.

Reply:
Readability is in the eye of the beholder.

Using raw `match` on Result or Option for such transformations is not really idiomatic, most Rustaceans would use and expect `unwrap_or` or `unwrap_or_else` here. My personal take is that if I `see unwrap_or*` I know that the `Ok` path is the expected, usual one, which my brain doesn't immediately see with a raw `match`. So, with that in mind, the first draft would give you this:

```rust

        let fmt_time = OffsetDateTime::now_utc()
            .format(format_description!(
                "[day]/[month]/[year] [hour]:[minute]:[second]"
            ))
            .unwrap_or(|_| "00/00/0000 00:00:00".into());

        eprintln!("{} - {}: {}", fmt_time, record.level(), record.args());
```

However, this code will give you the following `clippy` warning (done with the `rust-clippy` archive package):

```
warning: use of `unwrap_or` followed by a function call
 --> src/main.rs:9:10
  |
9 |         .unwrap_or("00/00/0000 00:00:00".into());
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| "00/00/0000 00:00:00".into())`
  |
  = note: `#[warn(clippy::or_fun_call)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
```

I'm actually surprised that `cargo clippy --fix` didn't fix this automatically! Here's the fixed version, gone through `cargo fmt`.

```rust
        let fmt_time = OffsetDateTime::now_utc()
            .format(format_description!(
                "[day]/[month]/[year] [hour]:[minute]:[second]"
            ))
            .unwrap_or_else(|_| "00/00/0000 00:00:00".into());

        eprintln!("{} - {}: {}", fmt_time, record.level(), record.args());
```

So, what you see as a micro-optimization is just the natural consequence of the idiomatic use of Result methods + avoidance of a well-known anti-pattern.

Now, if you really want to go into the weeds of micro-optimization, you could even avoid the allocation in the fallback case (**please don't use this** ;-) ):

```rust
    let fmt_time = OffsetDateTime::now_utc()
        .format(format_description!(
            "[day]/[month]/[year] [hour]:[minute]:[second]"
        ));
    // temp values don't outlive the statement they're used from, hence needing to assign the initial Result<String, _> to a named variable.
    // That's OK though, name shadowing is embraced within Rust (maybe this is one of the reasons for it?)
    let fmt_time = fmt_time
        .as_ref()
        .map(|s| s.as_str())
        .unwrap_or("00/00/0000 00:00:00");
```

Again: please don't use that last one.